elasticdl/pkg/ps/server.go

```diff
@@ -159,14 +159,22 @@ func (s *Server) PullEmbeddingVectors(ctx context.Context, in *proto.PullEmbeddi
 }
 
 // PushGradients push gradients to server
-func (s *Server) PushGradients(ctx context.Context, in *proto.Model) (*proto.PushGradientsResponse, error) {
+func (s *Server) PushGradients(ctx context.Context, in *proto.PushGradientsRequest) (*proto.PushGradientsResponse, error) {
 	// TODO: only support async now
 	var lrMultiplier = float32(1.0)
-	if s.lrStalenessModulation && s.Model.Version > in.Version {
-		staleness := s.Model.Version - in.Version
+	if s.lrStalenessModulation && s.Model.Version > in.Gradients.Version {
+		staleness := s.Model.Version - in.Gradients.Version
 		lrMultiplier = lrMultiplier / float32(staleness)
 	}
-	err := s.Opt.ApplyGradients(in, s.Model, lrMultiplier)
+	s.Opt.SetLR(in.LearningRate)
```

Comment:
When multiple workers are PushGradients to PS, `s.Opt.lr` may be overwritten among threads.
Probably add a lr argument in `s.Opt.ApplyGradients`. If set (>0), use it. Otherwise, use the default lr value in s.Opt.

Reply:
Great idea. Fix it.